### PR TITLE
Add Install Referrer Support for Multiple App Stores

### PR DESCRIPTION
**Pull request by gdeluna-branch**
## Reference
SDK-1173 -- [Android] Add Samsung App Store Referrer API Support
SDK-1322 -- [Android] Add Huawei App Store Referrer API Support
SDK-1323 -- [Android] Add Xiaomi GetApps Referrer API Support

## Description
This change provides Install Referrer support for Samsung's Galaxy Store, Huawei's AppGallery, and Xiaomi's GetApps app distribution platforms.

The repo now has a reference to the Huawei dependency but only for compilation via the Gradle `compileOnly` dependency configuration.

What is compileOnly?
https://developer.android.com/studio/build/dependencies#dependency_configurations

`Gradle adds the dependency to the compile classpath only (that is, it is not added to the build output). This is useful when you're creating an Android module and you need the dependency during compilation, but it's optional to have it present at runtime.`

For example, a developer only distributing their app on the Google Play Store will not see any Huawei APIs loaded into their APK. The other stores are currently implemented via reflection so no dependency is referenced. In other words, there is no change for the vast majority of integrators.

If they decide to additionally release on Huawei AppGallery, they need only provide the dependency (and declare the Huawei Maven repo to resolve it). Same with the other stores.

This removes most of the complexity resolving, proxying classes and invoking methods with reflection. However there is still single runtime check using reflection with a hardcoded classname to see if a given dependency is imported, per store.

Update:
For the not yet released apis reflection will be used until publicly accessible dependencies can be used for compilation only.

## Testing Instructions
Devices tested for this project:
HUAWEI SNE-LX3 - aka Huawei Mate 20 Lite
 - Huawei install referrer testing was done using public documentation to write out a test install referrer to a test package name.
 - They provide a test project here https://github.com/huaweicodelabs/Ads 

Xiaomi M2006C3LG - aka Xiaomi Redmi 9A
- This requires some collaboration with a Xiaomi representative.

Samsung SM-G950U aka - Galaxy S8
- Similar to Google Play install referrer testing, invoke the market app with referrer string then install your own package to read it out.

## Risk Assessment [`LOW`]

- [ ] I, the PR creator, have tested — integration, unit, or otherwise — this code.

## Reviewer Checklist (To be checked off by the reviewer only)

- [ ] JIRA Ticket is referenced in PR title.
- Correctness & Style
    - [ ] Conforms to [AOSP Style Guides](https://source.android.com/setup/contribute/code-style)
    - [ ] Mission critical pieces are documented in code and out of code as needed.
- [ ] Unit Tests reviewed and test issue sufficiently.
- [ ] Functionality was reviewed in QA independently by another engineer on the team.

### CODE REVIEW

File: Branch-SDK/src/main/java/io/branch/referral/StoreReferrer.java [hunk 1]
```diff
@@ -136,26 +112,26 @@ private static void processReferrerInfo(Context context, String rawReferrerStrin
                 if (referrerMap.containsKey(Defines.Jsonkey.LinkClickID.getKey())) {
```
Comment (jf-branch) [inline Branch-SDK/src/main/java/io/branch/referral/StoreReferrer.java:112]: @gdeluna-branch this applies to all App store referrers right? and not just google?

Comment (gdeluna-branch) [reply Branch-SDK/src/main/java/io/branch/referral/StoreReferrer.java:112]: That's correct, each store if called will go through the same flow as before

**Comment by gdeluna-branch** [comment]
Add mock tests, screenshots internally.

**Comment by echo-branch** [comment]
The changes look good and the unit tests pass for me. Thanks for the demo as well.

**Review by agerstner-branch — APPROVED** [review]
LGTM, good luck!

**Pull request merged**